googlemock/src/gmock-spec-builders.cc

```diff
@@ -632,6 +633,24 @@ bool Mock::VerifyAndClear(void* mock_obj)
   return VerifyAndClearExpectationsLocked(mock_obj);
 }
 
+// Set name for mock. Will be used in output.
+// Usefull when multiple instances of same mock is required.
+bool Mock::SetMockName(void* mock_obj, const std::string& mock_name)
+    GTEST_LOCK_EXCLUDED_(internal::g_gmock_mutex) {
+  internal::MutexLock l(&internal::g_gmock_mutex);
+
+  g_mock_object_registry.states()[mock_obj].name = mock_name;
+  return true;
+}
+
+// Returns mock name which was setted with SetMockName
+std::string Mock::GetMockName(const void* mock_obj)
+    GTEST_LOCK_EXCLUDED_(internal::g_gmock_mutex) {
+  internal::MutexLock l(&internal::g_gmock_mutex);
+  if (g_mock_object_registry.states().count(mock_obj) == 0) return "";
```

Comment:
We can use `!g_mock_object_registry.states().count(mock_obj)`

Reply:
I would prefer current one, it is more explicit as for me